virttest/guest_agent.py

```diff
@@ -730,3 +725,32 @@ def fsthaw(self, check_status=True):
                     raise
             return ret
         return -1
+
+    @error_context.context_aware
+    def file_operate(self, operate, **kwargs):
+        """
+        file operation function,
+        :param operate: open,write,read,flush,seek and flush
+        :param kwargs: optional keyword arguments including but not limited to below
+        :keyword Args:
+                handle: file handle returned by guest-file-open
+                mode: optional open mode, "r" is the default value.
+                content: content to write
+                count: optional,maximum number of bytes to read (default is 4KB)
+                offset: offset of whence
+                whence: 0,file beginning;
+                        1,file current position;
+                        2,file end
+        :return: The command's output
+        """
+        cmd = "guest-file-%s" % operate
+        if operate == "read" and kwargs["count"] is None:
+            kwargs.pop("count")
+        if operate == "write":
+            con = kwargs["content"]
+            kwargs.pop("content")
+            args = {"buf-b64": base64.b64encode(con.encode()).decode()}
+            kwargs.update(args)
+
+        if self.check_has_command(cmd):
+            return self.cmd(cmd=cmd, args=kwargs)
```

Comment:
Should we raise error for invalid command here? Or remove `self.check_has_command`, let `self.cmd()` handle invalid input itself?

Reply:
I think self.check_has_command() is enough to check the invalid command,it will raise VAgentCmdNotSupportedError. And  judge whether the command is invalid,we would better to use guest-info which is just what self.check_has_command() does.

Reply:
@xiagao Right, fair enough.